Index: src/rec/aSubRecord.c
===================================================================
--- src/rec/aSubRecord.c	(revision 19727)
+++ src/rec/aSubRecord.c	(revision 19728)
@@ -266,6 +266,7 @@
         return 0;
 
     prec->pact = TRUE;
+    recGblGetTimeStamp(prec);
 
     /* Push the output link values */
     if (!status) {
@@ -276,7 +277,6 @@
                 (&prec->neva)[i]);
     }
 
-    recGblGetTimeStamp(prec);
     monitor(prec);
     recGblFwdLink(prec);
     prec->pact = FALSE;
