diff --git a/src/ca/tcpiiu.cpp b/src/ca/tcpiiu.cpp
index 377e3ac..4770094 100644
--- a/src/ca/tcpiiu.cpp
+++ b/src/ca/tcpiiu.cpp
@@ -1456,9 +1456,10 @@ void tcpiiu::readNotifyRequest ( epicsGuard < epicsMutex > & guard,
     else {
         maxBytes = MAX_TCP;
     }
-    arrayElementCount maxElem = 
-        ( maxBytes - dbr_size[dataType] ) / dbr_value_size[dataType];
+    arrayElementCount elemSize  = dbr_value_size[dataType];
+    arrayElementCount maxElem   = ( maxBytes - dbr_size[dataType] ) / elemSize;
     if ( nElem > maxElem ) {
+        errlogPrintf( "CA Read Error: data size=%lu, max=%lu\n", nElem * elemSize, maxElem * elemSize );
         throw cacChannel::msgBodyCacheTooSmall ();
     }
     if (nElem == 0 && !CA_V413(this->minorProtocolVersion))
@@ -1569,8 +1570,10 @@ void tcpiiu::subscriptionRequest (
     }
     unsigned dataType = subscr.getType ( guard );
     // data type bounds checked when sunscription created
-    arrayElementCount maxElem = ( maxBytes - dbr_size[dataType] ) / dbr_value_size[dataType];
+    arrayElementCount elemSize  = dbr_value_size[dataType];
+    arrayElementCount maxElem   = ( maxBytes - dbr_size[dataType] ) / elemSize;
     if ( nElem > maxElem ) {
+        errlogPrintf( "CA Subscribe Error: data size=%lu, max=%lu\n", nElem * elemSize, maxElem * elemSize );
         throw cacChannel::msgBodyCacheTooSmall ();
     }
     comQueSendMsgMinder minder ( this->sendQue, guard );
@@ -1616,8 +1619,10 @@ void tcpiiu::subscriptionUpdateRequest (
     }
     unsigned dataType = subscr.getType ( guard );
     // data type bounds checked when subscription constructed
-    arrayElementCount maxElem = ( maxBytes - dbr_size[dataType] ) / dbr_value_size[dataType];
+    arrayElementCount elemSize  = dbr_value_size[dataType];
+    arrayElementCount maxElem   = ( maxBytes - dbr_size[dataType] ) / elemSize;
     if ( nElem > maxElem ) {
+        errlogPrintf( "CA Subscription Update Error: data size=%lu, max=%lu\n", nElem * elemSize, maxElem * elemSize );
         throw cacChannel::msgBodyCacheTooSmall ();
     }
     comQueSendMsgMinder minder ( this->sendQue, guard );
