diff --git a/src/ioc/db/dbCa.c b/src/ioc/db/dbCa.c
index 3f67269..d2b5a97 100644
--- a/src/ioc/db/dbCa.c
+++ b/src/ioc/db/dbCa.c
@@ -377,6 +377,7 @@ long dbCaPutLinkCallback(struct link *plink,short dbrType,
     long   status = 0;
     short  link_action = 0;
 
+    assert(plink->type == CA_LINK);
     assert(pca);
     /* put the new value in */
     epicsMutexMustLock(pca->lock);
@@ -468,6 +469,26 @@ void dbCaScanFwdLink(struct link *plink) {
         dbCaPutLink(plink, DBR_SHORT, &fwdLinkValue, 1);
 }
 
+int dbCaHasReadAccess(const struct link *plink)
+{
+    caLink *pca;
+
+    if (!plink || plink->type != CA_LINK) return FALSE;
+    pca = (caLink *)plink->value.pv_link.pvt;
+    if (!pca || !pca->chid) return FALSE;
+    return pca->hasReadAccess;
+}
+
+int dbCaHasWriteAccess(const struct link *plink)
+{
+    caLink *pca;
+
+    if (!plink || plink->type != CA_LINK) return FALSE;
+    pca = (caLink *)plink->value.pv_link.pvt;
+    if (!pca || !pca->chid) return FALSE;
+    return pca->hasWriteAccess;
+}
+
 #define pcaGetCheck \
     assert(plink); \
     if (plink->type != CA_LINK) return -1; \
diff --git a/src/ioc/db/dbCa.h b/src/ioc/db/dbCa.h
index cfdf732..e5ffad0 100644
--- a/src/ioc/db/dbCa.h
+++ b/src/ioc/db/dbCa.h
@@ -42,6 +42,8 @@ epicsShareFunc long dbCaPutLinkCallback(struct link *plink,
 #define dbCaPutLink(plink, dbrType, pbuffer, nRequest) \
     dbCaPutLinkCallback((plink), (dbrType), (pbuffer), (nRequest), 0, 0)
 epicsShareFunc int dbCaIsLinkConnected(const struct link *plink);
+epicsShareFunc int dbCaHasReadAccess(  const struct link *plink);
+epicsShareFunc int dbCaHasWriteAccess( const struct link *plink);
 epicsShareFunc void dbCaScanFwdLink(struct link *plink);
 
 /* The following are available after the link is connected*/
