diff --git a/src/libCom/osi/os/posix/osdEvent.c b/src/libCom/osi/os/posix/osdEvent.c
index dd3923c..76584ab 100644
--- a/src/libCom/osi/os/posix/osdEvent.c
+++ b/src/libCom/osi/os/posix/osdEvent.c
@@ -27,8 +27,14 @@
 #include "epicsAssert.h"
 
 /* Until these can be demonstrated to work leave them undefined*/
+/* We need this for linux; so we will try them now.
 #undef _POSIX_THREAD_PROCESS_SHARED
 #undef _POSIX_THREAD_PRIO_INHERIT
+*/
+#define _POSIX_THREAD_PROCESS_SHARED
+#define _POSIX_THREAD_PRIO_INHERIT
+
+
 
 typedef struct epicsEventOSD {
     pthread_mutex_t     mutex;
diff --git a/src/libCom/osi/os/posix/osdMutex.c b/src/libCom/osi/os/posix/osdMutex.c
index 064b82f..8f54dff 100644
--- a/src/libCom/osi/os/posix/osdMutex.c
+++ b/src/libCom/osi/os/posix/osdMutex.c
@@ -49,8 +49,14 @@ static int mutexLock(pthread_mutex_t *id)
 
 /* Until these can be demonstrated to work leave them undefined*/
 /* On solaris 8 _POSIX_THREAD_PRIO_INHERIT fails*/
+/* We need this for linux; so we will try them now.
 #undef _POSIX_THREAD_PROCESS_SHARED
 #undef _POSIX_THREAD_PRIO_INHERIT
+*/
+
+
+#define _POSIX_THREAD_PROCESS_SHARED
+#define _POSIX_THREAD_PRIO_INHERIT
 
 /* Two completely different implementations are provided below
  * If support is available for PTHREAD_MUTEX_RECURSIVE then
@@ -136,6 +142,7 @@ void epicsMutexOsdShow(struct epicsMutexOSD * pmutex, unsigned int level)
 }
 
 #else /*defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE)>=500 */
+#error "PTHREAD_MUTEX_RECURSIVE is not allowed... Sorry about it :(" 
 
 typedef struct epicsMutexOSD {
     pthread_mutexattr_t mutexAttr;
