Index: src/gdd/gddAppTable.cc
===================================================================
--- src/gdd/gddAppTable.cc	(revision 26163)
+++ src/gdd/gddAppTable.cc	(working copy)
@@ -436,8 +436,10 @@
 		}
 
 		// fprintf(stderr,"Adding DD to free_list %d\n",app);
+		attr_table[group][app].sem.lock ();
 		dd->setNext(attr_table[group][app].free_list);
 		attr_table[group][app].free_list=dd;
+		attr_table[group][app].sem.unlock ();
 	}
 	else if (attr_table[group][app].type==gddApplicationTypeNormal)
 	{
