--- configure/os/CONFIG_SITE.Common.linuxRT_glibc-x86_64	1969-12-31 16:00:00.000000000 -0800
+++ configure/os/CONFIG_SITE.Common.linuxRT_glibc-x86_64	2015-04-02 09:57:44.974265952 -0700
@@ -0,0 +1,36 @@
+# Build types
+VALID_BUILDS = Ioc
+
+# Location of linuxRT GNU ToolChain
+#LINUX_RT_HOME=/nfs/slac/g/lcls/build/linuxRT
+#LINUX_RT_HOME=/afs/slac/package/linuxRT
+#LINUX_RT_HOME=/usr/local/common
+#LINUX_RT_HOME=/home/ernesto/packages
+LINUX_RT_HOME=/reg/g/pcds/package/linuxRT
+
+#GNU_DIR=$(LINUX_RT_HOME)/buildroot/buildroot-2012.02-x86/output/host/usr/
+#GNU_DIR=$(LINUX_RT_HOME)/buildroot-2014.08/host/$(EPICS_HOST_ARCH)/usr
+GNU_DIR=$(LINUX_RT_HOME)/buildroot-2014.11/host/$(EPICS_HOST_ARCH)/x86_64-glibc-linux/usr
+
+# Cross Target Architecture
+GNU_TARGET=x86_64-linux
+
+# Could substitute prefix of rpath here
+#SHRLIBDIR_RPATH_LDFLAGS_YES = $(SHRLIB_DEPLIB_DIRS:%=-Wl,-rpath,%)
+# Only static build is supported at this time.
+SHARED_LIBRARIES=NO
+STATIC_BUILD=YES
+
+# =====================================================================
+# Recommended by Till Strauman:
+# For linuxRT-x86 Builds.
+# Let's build with optimization but have debug symbols for debugging:
+# =====================================================================
+OPT_CFLAGS_YES = -O3 -g
+OPT_CXXFLAGS_YES = -O3 -g
+
+# =================================================
+# Add support for readline: used by
+# EPICS IOC Shell for command line history, etc...
+#  ================================================
+COMMANDLINE_LIBRARY = READLINE

--- configure/os/CONFIG.Common.linuxRT_glibc-x86_64	1969-12-31 16:00:00.000000000 -0800
+++ configure/os/CONFIG.Common.linuxRT_glibc-x86_64	2015-04-02 11:13:42.214987885 -0700
@@ -0,0 +1,17 @@
+# CONFIG.Common.linuxRT-x86_64
+#
+# This file is maintained by the build community.
+#
+# Definitions for linuxRT-x86_64 target builds
+# linuxRT is really linux-rt with uClibc produced by Till Strauman of
+# SLAC National Laboratories
+# Sites may override these definitions and add other definitions
+# in CONFIG_SITE.Common.linuxRT-x86_64
+#-------------------------------------------------------
+
+# Include definitions common to all Unix targets
+include $(CONFIG)/os/CONFIG.Common.linuxCommon
+VALID_BUILDS = Host Ioc
+ARCH_CLASS = x86_64
+ARCH_DEP_CPPFLAGS += -D_X86_64_
+
