diff --git a/src/libCom/macLib/macCore.c b/src/libCom/macLib/macCore.c
index 3d024ad..be51a80 100644
--- a/src/libCom/macLib/macCore.c
+++ b/src/libCom/macLib/macCore.c
@@ -869,7 +869,7 @@ static void refer ( MAC_HANDLE *handle, MAC_ENTRY *entry, int level,
             if ( !handle->dirty ) {
                 /* copy the already-expanded value, and its error status! */
                 cpy2val( refentry->value, &v, valend );
-                entry->error = refentry->error;
+                if (refentry->error == TRUE) entry->error = TRUE;
             } else {
                 /* translate raw value */
                 const char *rv = refentry->rawval;
diff --git a/src/libCom/test/macLibTest.c b/src/libCom/test/macLibTest.c
index 1f00a6a..a80ef79 100644
--- a/src/libCom/test/macLibTest.c
+++ b/src/libCom/test/macLibTest.c
@@ -65,7 +65,7 @@ static void ovcheck(void)
 
 MAIN(macLibTest)
 {
-    testPlan(89);
+    testPlan(91);
 
     if (macCreateHandle(&h, NULL))
         testAbort("macCreateHandle() failed");
@@ -202,6 +202,8 @@ MAIN(macLibTest)
         /* STR1 = "VAL1" */
         /* STR2 = "VAL2" */
     check("${FOO}", " VAL2");
+    check("$(FOO)$(FOO1)", "!VAL2$(FOO1,undefined)");
+    check("$(FOO1)$(FOO)", "!$(FOO1,undefined)VAL2");
 
     macPutValue(h, "BAR","${FOO}");
         /* FOO = "${BAR}" */
