diff --git a/configure/RULES.Db b/configure/RULES.Db
index fdc886c..6636d9a 100644
--- a/configure/RULES.Db
+++ b/configure/RULES.Db
@@ -342,7 +342,7 @@ $(COMMON_DIR)/%.db$(RAW): %.sub-db
 	@$(MKMF) -m$(notdir $@)$(DEP) $(DBFLAGS) $@ $< $(TEMPLATE_FILENAME)
 	echo "$@ : $(TEMPLATE_FILENAME)" >> $(notdir $@)$(DEP)
 	echo "$(notdir $@)$(DEP): $(TEMPLATE_FILENAME)" >> $(notdir $@)$(DEP)
-	$(ECHO) "Inflating database from $<"
+	$(ECHO) "Inflating database from $< $(TEMPLATE_FILENAME)"
 	@$(RM) $@ $*.tmp
 	$(MSI) $(DBFLAGS) -S$< > $*.tmp
 	$(MV) $*.tmp $@
@@ -352,7 +352,7 @@ $(COMMON_DIR)/%.db$(RAW): %.template
 	@$(MKMF) -m$(notdir $@)$(DEP) $(DBFLAGS) $@ $< $(TEMPLATE_FILENAME)
 	$(ECHO) "Inflating database from $<"
 	@$(RM) $@ $*.tmp
-	$(MSI) $(DBFLAGS) $< > $*.tmp
+	$(MSI) $(DBFLAGS) -S$< > $*.tmp
 	$(MV) $*.tmp $@
 
 $(COMMON_DIR)/%.acf: %.acs
diff --git a/src/tools/mkmf.pl b/src/tools/mkmf.pl
index 4334d29..60cfd97 100644
--- a/src/tools/mkmf.pl
+++ b/src/tools/mkmf.pl
@@ -99,7 +99,7 @@ sub scanFile {
    print "Scanning file $file\n" if $debug;
    open FILE, $file or return;
    foreach $line ( <FILE> ) {
-      $incfile = findNextIncName($line,$file=~/\.substitutions$/);
+      $incfile = findNextIncName($line,$file=~/\.sub/);
       next if !$incfile;
       next if $output{$incfile};   
       push @includes,$incfile;
