diff --git a/startup/EpicsHostArch b/startup/EpicsHostArch
index 8861ac5..c4df266 100755
--- a/startup/EpicsHostArch
+++ b/startup/EpicsHostArch
@@ -24,6 +24,17 @@ case $sysname in
     Linux )
         os=linux
         cpu=`uname -m`
+        release=`uname -r`
+		case $release in
+			*el6* )
+				os=rhel6 ;;
+			*el7* )
+				os=rhel7 ;;
+			2.6.35.13-rt )
+				os=linux ;;
+			3.14.12-rt9 | 3.18.11-rt7 )
+				os=linuxRT_glibc ;;
+        esac
         case $cpu in
             i386 | i486 | i586 | i686 )
                 cpu=x86 ;;
diff --git a/startup/EpicsHostArch.pl b/startup/EpicsHostArch.pl
index e6babf6..aa4ae1c 100755
--- a/startup/EpicsHostArch.pl
+++ b/startup/EpicsHostArch.pl
@@ -16,6 +16,12 @@ eval 'exec perl -S $0 ${1+"$@"}'  # -*- Mode: perl -*-
 use Config;
 use POSIX;
 
+use Config qw( config_sh myconfig );
+#print config_sh();
+#print myconfig();
+#print "Config{'archname'}=".$Config{'archname'}."\n";
+#print "Config{'archname64'}=".$Config{'archname64'}."\n";
+
 $suffix="";
 $suffix="-".$ARGV[0] if ($ARGV[0] ne "");
 
@@ -24,11 +30,28 @@ print "$EpicsHostArch$suffix";
 
 sub GetEpicsHostArch { # no args
     $arch=$Config{'archname'};
+    #print "Config{'archname'}=".$arch."\n";
     if ($arch =~ /sun4-solaris/)        { return "solaris-sparc";
     } elsif ($arch =~ m/i86pc-solaris/) { return "solaris-x86";
-    } elsif ($arch =~ m/i[3-6]86-linux/){ return "linux-x86";
-    } elsif ($arch =~ m/x86_64-linux/)  { return "linux-x86_64";
     } elsif ($arch =~ m/arm-linux/)     { return "linux-arm";
+    } elsif ($arch =~ m/linux/)        {
+            my($kernel, $hostname, $release, $version, $cpu) = POSIX::uname();
+    		#print "kernel=".$kernel."\n";
+    		#print "hostname=".$hostname."\n";
+    		#print "release=".$release."\n";
+    		#print "version=".$version."\n";
+    		#print "cpu=".$cpu."\n";
+            if ($cpu =~ m/i686/)			{ return "linux-x86";  }
+            if ($cpu =~ m/x86_64/)	{
+				if ($release =~ m/el5/)     { return "linux-x86_64";  }
+				elsif ($release =~ m/2.6.35.13-rt/)  { return "linux-x86_64"; }
+				elsif ($release =~ m/3.14.12-rt9/)  { return "linuxRT_glibc-x86_64"; }
+				elsif ($release =~ m/3.18.11-rt7/)  { return "linuxRT_glibc-x86_64"; }
+				elsif ($release =~ m/el6/)  { return "rhel6-x86_64"; }
+				elsif ($release =~ m/el7/)  { return "rhel7-x86_64"; }
+				elsif ($release =~ m/2.6.26.1/)  { return "linux-x86_64"; }
+			}
+            else							{ return "unsupported"; }
     } elsif ($arch =~ m/MSWin32-x86/)   { return "win32-x86";
     } elsif ($arch =~ m/MSWin32-x64/)   { return "windows-x64";
     } elsif ($arch =~ m/cygwin/)        {
diff --git a/startup/win32.bat b/startup/win32.bat
old mode 100644
new mode 100755
