diff --git a/configure/os/CONFIG.Common.rhel6-x86_64 b/configure/os/CONFIG.Common.rhel6-x86_64
new file mode 100644
index 0000000..7dd4453
--- /dev/null
+++ b/configure/os/CONFIG.Common.rhel6-x86_64
@@ -0,0 +1,24 @@
+# CONFIG.Common.rhel6-x86_64
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel6-x86_64 target builds
+# Sites may override these definitions in CONFIG_SITE.Common.rhel6-x86_64
+#-------------------------------------------------------
+
+# Include definitions common to all linux targets
+include $(CONFIG)/os/CONFIG.Common.linuxCommon
+
+ARCH_CLASS = x86_64
+
+ARCH_DEP_CPPFLAGS += -D_X86_64_
+ARCH_DEP_CFLAGS += -m64
+ARCH_DEP_LDFLAGS += -m64
+
+OS_CLASS = Linux
+
+# If your crosscompiler name has a GNU target prefix like <gnutarget>-gcc,
+# e.g. x86_64-redhat-linux-gcc, put a GNU_TARGET definition in
+# CONFIG_SITE.<host>.rhel6-x86_64 file, e.g. GNU_TARGET=x86_64-redhat-linux
+
diff --git a/configure/os/CONFIG.Common.rhel6-x86_64-debug b/configure/os/CONFIG.Common.rhel6-x86_64-debug
new file mode 100644
index 0000000..1aff2c9
--- /dev/null
+++ b/configure/os/CONFIG.Common.rhel6-x86_64-debug
@@ -0,0 +1,13 @@
+# CONFIG.Common.rhel6-x86_64-debug
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel6-x86_64 with debug compiler flags
+# Sites may override these definitions in CONFIG_SITE.Common.rhel6-x86_64-debug
+#-------------------------------------------------------
+
+# Include definitions common to all rhel6-x86_64  target archs
+include $(CONFIG)/os/CONFIG.Common.rhel6-x86_64
+
+HOST_OPT=NO
diff --git a/configure/os/CONFIG.Common.rhel7-x86_64 b/configure/os/CONFIG.Common.rhel7-x86_64
new file mode 100644
index 0000000..bca002b
--- /dev/null
+++ b/configure/os/CONFIG.Common.rhel7-x86_64
@@ -0,0 +1,24 @@
+# CONFIG.Common.rhel7-x86_64
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel7-x86_64 target builds
+# Sites may override these definitions in CONFIG_SITE.Common.rhel7-x86_64
+#-------------------------------------------------------
+
+# Include definitions common to all linux targets
+include $(CONFIG)/os/CONFIG.Common.linuxCommon
+
+ARCH_CLASS = x86_64
+
+ARCH_DEP_CPPFLAGS += -D_X86_64_
+ARCH_DEP_CFLAGS += -m64
+ARCH_DEP_LDFLAGS += -m64
+
+OS_CLASS = Linux
+
+# If your crosscompiler name has a GNU target prefix like <gnutarget>-gcc,
+# e.g. x86_64-redhat-linux-gcc, put a GNU_TARGET definition in
+# CONFIG_SITE.<host>.rhel7-x86_64 file, e.g. GNU_TARGET=x86_64-redhat-linux
+
diff --git a/configure/os/CONFIG.Common.rhel7-x86_64-debug b/configure/os/CONFIG.Common.rhel7-x86_64-debug
new file mode 100644
index 0000000..d1f8f4e
--- /dev/null
+++ b/configure/os/CONFIG.Common.rhel7-x86_64-debug
@@ -0,0 +1,13 @@
+# CONFIG.Common.rhel7-x86_64-debug
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel7-x86_64 with debug compiler flags
+# Sites may override these definitions in CONFIG_SITE.Common.rhel7-x86_64-debug
+#-------------------------------------------------------
+
+# Include definitions common to all rhel7-x86_64  target archs
+include $(CONFIG)/os/CONFIG.Common.rhel7-x86_64
+
+HOST_OPT=NO
diff --git a/configure/os/CONFIG.rhel6-x86_64-debug.Common b/configure/os/CONFIG.rhel6-x86_64-debug.Common
new file mode 100644
index 0000000..7b90f32
--- /dev/null
+++ b/configure/os/CONFIG.rhel6-x86_64-debug.Common
@@ -0,0 +1,15 @@
+# CONFIG.rhel6-x86_64-debug.Common
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel6-x86_64 debug with debug compiler flags
+# Sites may override these definitions in CONFIG_SITE.rhel6-x86_64-debug.Common
+#-------------------------------------------------------
+
+#Include definitions common to rhel6-x86_64 hosts
+include $(CONFIG)/os/CONFIG.rhel6-x86_64.Common
+
+# Make all crosscompiler builds debug builds
+#CROSS_OPT=NO
+
diff --git a/configure/os/CONFIG.rhel6-x86_64-debug.rhel6-x86_64-debug b/configure/os/CONFIG.rhel6-x86_64-debug.rhel6-x86_64-debug
new file mode 100644
index 0000000..a7747cd
--- /dev/null
+++ b/configure/os/CONFIG.rhel6-x86_64-debug.rhel6-x86_64-debug
@@ -0,0 +1,14 @@
+# CONFIG.rhel6-x86_64-debug.rhel6-x86_64-debug
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel6-x86_64 host - rhel6-x86_64 target build with debug compiler flags
+# Sites may override these definitions in CONFIG_SITE.rhel6-x86_64-debug.rhel6-x86_64-debug
+#-------------------------------------------------------
+
+include $(CONFIG)/os/CONFIG.rhel6-x86_64.rhel6-x86_64
+
+# Removes -O optimization and adds -g compile option
+HOST_OPT=NO
+
diff --git a/configure/os/CONFIG.rhel6-x86_64.Common b/configure/os/CONFIG.rhel6-x86_64.Common
new file mode 100644
index 0000000..eb335ac
--- /dev/null
+++ b/configure/os/CONFIG.rhel6-x86_64.Common
@@ -0,0 +1,13 @@
+# CONFIG.rhel6-x86_64.Common
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel6-x86_64 host builds
+# Sites may override these definitions in CONFIG_SITE.rhel6-x86_64.Common
+#-------------------------------------------------------
+
+#Include definitions common to unix hosts
+include $(CONFIG)/os/CONFIG.UnixCommon.Common
+
+WIND_HOST_TYPE = x86-linux
diff --git a/configure/os/CONFIG.rhel6-x86_64.rhel6-x86_64 b/configure/os/CONFIG.rhel6-x86_64.rhel6-x86_64
new file mode 100644
index 0000000..7d8ae01
--- /dev/null
+++ b/configure/os/CONFIG.rhel6-x86_64.rhel6-x86_64
@@ -0,0 +1,12 @@
+#  CONFIG.rhel6-x86_64.rhel6-x86_64
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+#
+# Definitions for rhel6-x86_64 host - rhel6-x86_64 target builds
+# Sites may override these definitions in CONFIG_SITE.rhel6-x86_64.rhel6-x86_64
+#-------------------------------------------------------
+
+# Include common gnu compiler definitions
+include $(CONFIG)/os/CONFIG.linux-x86.linux-x86
+
+GNU_LIB = $(GNU_DIR)/lib64
diff --git a/configure/os/CONFIG.rhel6-x86_64.rhel6-x86_64-debug b/configure/os/CONFIG.rhel6-x86_64.rhel6-x86_64-debug
new file mode 100644
index 0000000..f5a4614
--- /dev/null
+++ b/configure/os/CONFIG.rhel6-x86_64.rhel6-x86_64-debug
@@ -0,0 +1,15 @@
+# CONFIG.rhel6-x86_64.rhel6-x86_64-debug
+#
+# Revision-Id: jba@aps.anl.gov-20101115163116-3xboj0bh17bood7s
+# This file is maintained by the build community.
+#
+# Definitions for rhel6-x86_64 host - rhel6-x86_64-debug target build with debug compiler flags
+# Sites may override these definitions in CONFIG_SITE.rhel6-x86_64.rhel6-x86_64-debug
+#-------------------------------------------------------
+
+-include $(CONFIG)/os/CONFIG.rhel6-x86_64.rhel6-x86_64
+-include $(CONFIG)/os/CONFIG_SITE.Common.rhel6-x86_64
+-include $(CONFIG)/os/CONFIG_SITE.rhel6-x86_64.rhel6-x86_64
+
+BUILD_CLASS=HOST
+HOST_OPT = NO
diff --git a/configure/os/CONFIG.rhel7-x86_64-debug.Common b/configure/os/CONFIG.rhel7-x86_64-debug.Common
new file mode 100644
index 0000000..bbe7981
--- /dev/null
+++ b/configure/os/CONFIG.rhel7-x86_64-debug.Common
@@ -0,0 +1,15 @@
+# CONFIG.rhel7-x86_64-debug.Common
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel7-x86_64 debug with debug compiler flags
+# Sites may override these definitions in CONFIG_SITE.rhel7-x86_64-debug.Common
+#-------------------------------------------------------
+
+#Include definitions common to rhel7-x86_64 hosts
+include $(CONFIG)/os/CONFIG.rhel7-x86_64.Common
+
+# Make all crosscompiler builds debug builds
+#CROSS_OPT=NO
+
diff --git a/configure/os/CONFIG.rhel7-x86_64-debug.rhel7-x86_64-debug b/configure/os/CONFIG.rhel7-x86_64-debug.rhel7-x86_64-debug
new file mode 100644
index 0000000..fef75ae
--- /dev/null
+++ b/configure/os/CONFIG.rhel7-x86_64-debug.rhel7-x86_64-debug
@@ -0,0 +1,14 @@
+# CONFIG.rhel7-x86_64-debug.rhel7-x86_64-debug
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel7-x86_64 host - rhel7-x86_64 target build with debug compiler flags
+# Sites may override these definitions in CONFIG_SITE.rhel7-x86_64-debug.rhel7-x86_64-debug
+#-------------------------------------------------------
+
+include $(CONFIG)/os/CONFIG.rhel7-x86_64.rhel7-x86_64
+
+# Removes -O optimization and adds -g compile option
+HOST_OPT=NO
+
diff --git a/configure/os/CONFIG.rhel7-x86_64.Common b/configure/os/CONFIG.rhel7-x86_64.Common
new file mode 100644
index 0000000..66dcbd9
--- /dev/null
+++ b/configure/os/CONFIG.rhel7-x86_64.Common
@@ -0,0 +1,13 @@
+# CONFIG.rhel7-x86_64.Common
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Definitions for rhel7-x86_64 host builds
+# Sites may override these definitions in CONFIG_SITE.rhel7-x86_64.Common
+#-------------------------------------------------------
+
+#Include definitions common to unix hosts
+include $(CONFIG)/os/CONFIG.UnixCommon.Common
+
+WIND_HOST_TYPE = x86-linux
diff --git a/configure/os/CONFIG.rhel7-x86_64.rhel7-x86_64 b/configure/os/CONFIG.rhel7-x86_64.rhel7-x86_64
new file mode 100644
index 0000000..ea56c5a
--- /dev/null
+++ b/configure/os/CONFIG.rhel7-x86_64.rhel7-x86_64
@@ -0,0 +1,12 @@
+#  CONFIG.rhel7-x86_64.rhel7-x86_64
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+#
+# Definitions for rhel7-x86_64 host - rhel7-x86_64 target builds
+# Sites may override these definitions in CONFIG_SITE.rhel7-x86_64.rhel7-x86_64
+#-------------------------------------------------------
+
+# Include common gnu compiler definitions
+include $(CONFIG)/os/CONFIG.linux-x86.linux-x86
+
+GNU_LIB = $(GNU_DIR)/lib64
diff --git a/configure/os/CONFIG.rhel7-x86_64.rhel7-x86_64-debug b/configure/os/CONFIG.rhel7-x86_64.rhel7-x86_64-debug
new file mode 100644
index 0000000..3a9be63
--- /dev/null
+++ b/configure/os/CONFIG.rhel7-x86_64.rhel7-x86_64-debug
@@ -0,0 +1,15 @@
+# CONFIG.rhel7-x86_64.rhel7-x86_64-debug
+#
+# Revision-Id: jba@aps.anl.gov-20101115163116-3xboj0bh17bood7s
+# This file is maintained by the build community.
+#
+# Definitions for rhel7-x86_64 host - rhel7-x86_64-debug target build with debug compiler flags
+# Sites may override these definitions in CONFIG_SITE.rhel7-x86_64.rhel7-x86_64-debug
+#-------------------------------------------------------
+
+-include $(CONFIG)/os/CONFIG.rhel7-x86_64.rhel7-x86_64
+-include $(CONFIG)/os/CONFIG_SITE.Common.rhel7-x86_64
+-include $(CONFIG)/os/CONFIG_SITE.rhel7-x86_64.rhel7-x86_64
+
+BUILD_CLASS=HOST
+HOST_OPT = NO
diff --git a/configure/os/CONFIG_SITE.Common.rhel6-x86_64 b/configure/os/CONFIG_SITE.Common.rhel6-x86_64
new file mode 100644
index 0000000..c762bd2
--- /dev/null
+++ b/configure/os/CONFIG_SITE.Common.rhel6-x86_64
@@ -0,0 +1,44 @@
+# CONFIG_SITE.Common.rhel6-x86_64
+#
+# Revision-Id: jba@aps.anl.gov-20130725154926-szg53grbkd982szq
+#
+# Site Specific definitions for rhel6-x86_64 target
+# Only the local epics system manager should modify this file
+
+# NOTE for SHARED_LIBRARIES: In most cases if this is set to YES the
+# shared libraries will be found automatically.  However if the .so
+# files are installed at a different path to their compile-time path
+# then in order to be found at runtime do one of these:
+# a) LD_LIBRARY_PATH must include the full absolute pathname to
+#    $(INSTALL_LOCATION)/lib/$(EPICS_HOST_ARCH) when invoking base
+#    executables.
+# b) Add the runtime path to SHRLIB_DEPLIB_DIRS and PROD_DEPLIB_DIRS, which
+#    will add the named directory to the list contained in the executables.
+# c) Add the runtime path to /etc/ld.so.conf and run ldconfig
+#    to inform the system of the shared library location.
+
+# Depending on your version of Linux you'll want one of the following
+# lines to enable command-line editing and history in iocsh.  If you're
+# not sure which, start with the top one and work downwards until the
+# build doesn't fail to link the readline library.  If none of them work,
+# comment them all out to build without readline support.
+
+# No other libraries needed (recent Fedora, Ubuntu etc.):
+#COMMANDLINE_LIBRARY = READLINE
+
+# Needs -lncurses (RHEL 5 etc.):
+COMMANDLINE_LIBRARY = READLINE_NCURSES
+
+# Needs -lcurses (older versions)
+#COMMANDLINE_LIBRARY = READLINE_CURSES
+
+
+# It makes sense to include debugging symbols even in optimized builds
+# in case you want to attach gdb to the process or examine a core-dump.
+# This does cost disk space, but not memory as debug symbols are not
+# loaded into RAM when the binary is loaded.
+OPT_CFLAGS_YES += -g
+OPT_CXXFLAGS_YES += -g
+
+# Support Package Architecture
+PKG_ARCH=x86_64-rhel6-gcc44-opt
diff --git a/configure/os/CONFIG_SITE.Common.rhel7-x86_64 b/configure/os/CONFIG_SITE.Common.rhel7-x86_64
new file mode 100644
index 0000000..7625768
--- /dev/null
+++ b/configure/os/CONFIG_SITE.Common.rhel7-x86_64
@@ -0,0 +1,44 @@
+# CONFIG_SITE.Common.rhel7-x86_64
+#
+# Revision-Id: jba@aps.anl.gov-20130725154926-szg53grbkd982szq
+#
+# Site Specific definitions for rhel7-x86_64 target
+# Only the local epics system manager should modify this file
+
+# NOTE for SHARED_LIBRARIES: In most cases if this is set to YES the
+# shared libraries will be found automatically.  However if the .so
+# files are installed at a different path to their compile-time path
+# then in order to be found at runtime do one of these:
+# a) LD_LIBRARY_PATH must include the full absolute pathname to
+#    $(INSTALL_LOCATION)/lib/$(EPICS_HOST_ARCH) when invoking base
+#    executables.
+# b) Add the runtime path to SHRLIB_DEPLIB_DIRS and PROD_DEPLIB_DIRS, which
+#    will add the named directory to the list contained in the executables.
+# c) Add the runtime path to /etc/ld.so.conf and run ldconfig
+#    to inform the system of the shared library location.
+
+# Depending on your version of Linux you'll want one of the following
+# lines to enable command-line editing and history in iocsh.  If you're
+# not sure which, start with the top one and work downwards until the
+# build doesn't fail to link the readline library.  If none of them work,
+# comment them all out to build without readline support.
+
+# No other libraries needed (recent Fedora, Ubuntu etc.):
+#COMMANDLINE_LIBRARY = READLINE
+
+# Needs -lncurses (RHEL 5 etc.):
+COMMANDLINE_LIBRARY = READLINE_NCURSES
+
+# Needs -lcurses (older versions)
+#COMMANDLINE_LIBRARY = READLINE_CURSES
+
+
+# It makes sense to include debugging symbols even in optimized builds
+# in case you want to attach gdb to the process or examine a core-dump.
+# This does cost disk space, but not memory as debug symbols are not
+# loaded into RAM when the binary is loaded.
+OPT_CFLAGS_YES += -g
+OPT_CXXFLAGS_YES += -g
+
+# Support Package Architecture
+PKG_ARCH=x86_64-rhel7-gcc44-opt
diff --git a/configure/os/CONFIG_SITE.rhel6-x86_64-debug.rhel6-x86_64-debug b/configure/os/CONFIG_SITE.rhel6-x86_64-debug.rhel6-x86_64-debug
new file mode 100644
index 0000000..e747541
--- /dev/null
+++ b/configure/os/CONFIG_SITE.rhel6-x86_64-debug.rhel6-x86_64-debug
@@ -0,0 +1,15 @@
+# CONFIG_SITE.rhel6-x86_64-debug.rhel6-x86_64-debug
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+#
+# Site specific override definitions for solaris-sparc host builds
+# Only the local epics system manager should modify this file
+
+
+
+#Prepares the object code to collect data for profiling with prof. 
+#PROFILE=YES
+
+#Compiles for profiling with the gprof profiler.
+#GPROF=YES
+
diff --git a/configure/os/CONFIG_SITE.rhel6-x86_64.Common b/configure/os/CONFIG_SITE.rhel6-x86_64.Common
new file mode 100644
index 0000000..ff124d6
--- /dev/null
+++ b/configure/os/CONFIG_SITE.rhel6-x86_64.Common
@@ -0,0 +1,19 @@
+# CONFIG_SITE.rhel6-x86_64.Common
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Site override definitions for rhel6-x86_64 host builds
+#-------------------------------------------------------
+
+#CROSS_COMPILER_TARGET_ARCHS = vxWorks-68040 solaris-sparc
+#CROSS_COMPILER_TARGET_ARCHS = vxWorks-68040
+#CROSS_COMPILER_TARGET_ARCHS = RTEMS-mvme2100
+
+# =====================================================================
+# Recommended by Till Strauman:
+# For Host Builds.
+# Let's build with optimization but have debug symbols for debugging:
+# =====================================================================
+OPT_CFLAGS_YES = -O3 -g
+OPT_CXXFLAGS_YES = -O3 -g
diff --git a/configure/os/CONFIG_SITE.rhel6-x86_64.UnixCommon b/configure/os/CONFIG_SITE.rhel6-x86_64.UnixCommon
new file mode 100644
index 0000000..bab8f7e
--- /dev/null
+++ b/configure/os/CONFIG_SITE.rhel6-x86_64.UnixCommon
@@ -0,0 +1,7 @@
+# CONFIG_SITE.rhel6-x86_64.UnixCommon
+
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+#
+# Site Specific configure override definitions
+# Only the local epics system manager should modify this file
+ 
diff --git a/configure/os/CONFIG_SITE.rhel6-x86_64.rhel6-x86_64 b/configure/os/CONFIG_SITE.rhel6-x86_64.rhel6-x86_64
new file mode 100644
index 0000000..a2ea749
--- /dev/null
+++ b/configure/os/CONFIG_SITE.rhel6-x86_64.rhel6-x86_64
@@ -0,0 +1,8 @@
+# CONFIG_SITE.rhel6-x86_64.rhel6-x86_64
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+#
+# Site specific definitions for rhel6-x86_64 host - rhel6-x86_64 target builds
+#-------------------------------------------------------
+
+
diff --git a/configure/os/CONFIG_SITE.rhel7-x86_64-debug.rhel7-x86_64-debug b/configure/os/CONFIG_SITE.rhel7-x86_64-debug.rhel7-x86_64-debug
new file mode 100644
index 0000000..7978936
--- /dev/null
+++ b/configure/os/CONFIG_SITE.rhel7-x86_64-debug.rhel7-x86_64-debug
@@ -0,0 +1,15 @@
+# CONFIG_SITE.rhel7-x86_64-debug.rhel7-x86_64-debug
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+#
+# Site specific override definitions for solaris-sparc host builds
+# Only the local epics system manager should modify this file
+
+
+
+#Prepares the object code to collect data for profiling with prof. 
+#PROFILE=YES
+
+#Compiles for profiling with the gprof profiler.
+#GPROF=YES
+
diff --git a/configure/os/CONFIG_SITE.rhel7-x86_64.Common b/configure/os/CONFIG_SITE.rhel7-x86_64.Common
new file mode 100644
index 0000000..e61ed18
--- /dev/null
+++ b/configure/os/CONFIG_SITE.rhel7-x86_64.Common
@@ -0,0 +1,19 @@
+# CONFIG_SITE.rhel7-x86_64.Common
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+# This file is maintained by the build community.
+#
+# Site override definitions for rhel7-x86_64 host builds
+#-------------------------------------------------------
+
+#CROSS_COMPILER_TARGET_ARCHS = vxWorks-68040 solaris-sparc
+#CROSS_COMPILER_TARGET_ARCHS = vxWorks-68040
+#CROSS_COMPILER_TARGET_ARCHS = RTEMS-mvme2100
+
+# =====================================================================
+# Recommended by Till Strauman:
+# For Host Builds.
+# Let's build with optimization but have debug symbols for debugging:
+# =====================================================================
+OPT_CFLAGS_YES = -O3 -g
+OPT_CXXFLAGS_YES = -O3 -g
diff --git a/configure/os/CONFIG_SITE.rhel7-x86_64.UnixCommon b/configure/os/CONFIG_SITE.rhel7-x86_64.UnixCommon
new file mode 100644
index 0000000..8fb0c28
--- /dev/null
+++ b/configure/os/CONFIG_SITE.rhel7-x86_64.UnixCommon
@@ -0,0 +1,7 @@
+# CONFIG_SITE.rhel7-x86_64.UnixCommon
+
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+#
+# Site Specific configure override definitions
+# Only the local epics system manager should modify this file
+ 
diff --git a/configure/os/CONFIG_SITE.rhel7-x86_64.rhel7-x86_64 b/configure/os/CONFIG_SITE.rhel7-x86_64.rhel7-x86_64
new file mode 100644
index 0000000..fc9d829
--- /dev/null
+++ b/configure/os/CONFIG_SITE.rhel7-x86_64.rhel7-x86_64
@@ -0,0 +1,8 @@
+# CONFIG_SITE.rhel7-x86_64.rhel7-x86_64
+#
+# Revision-Id: anj@aps.anl.gov-20101005192737-disfz3vs0f3fiixd
+#
+# Site specific definitions for rhel7-x86_64 host - rhel7-x86_64 target builds
+#-------------------------------------------------------
+
+
diff --git a/startup/EpicsHostArch b/startup/EpicsHostArch
index 8861ac5..7f54895 100755
--- a/startup/EpicsHostArch
+++ b/startup/EpicsHostArch
@@ -24,6 +24,15 @@ case $sysname in
     Linux )
         os=linux
         cpu=`uname -m`
+        release=`uname -r`
+		case $release in
+			*el6* )
+				os=rhel6 ;;
+			*el7* )
+				os=rhel7 ;;
+			*-rt* )
+				os=linuxRT ;;
+        esac
         case $cpu in
             i386 | i486 | i586 | i686 )
                 cpu=x86 ;;
diff --git a/startup/EpicsHostArch.pl b/startup/EpicsHostArch.pl
index 09f7ffd..3ecb66d 100755
--- a/startup/EpicsHostArch.pl
+++ b/startup/EpicsHostArch.pl
@@ -25,9 +25,18 @@ sub GetEpicsHostArch { # no args
     $arch=$Config{'archname'};
     if ($arch =~ /sun4-solaris/)        { return "solaris-sparc";
     } elsif ($arch =~ m/i86pc-solaris/) { return "solaris-x86";
-    } elsif ($arch =~ m/i[3-6]86-linux/){ return "linux-x86";
-    } elsif ($arch =~ m/x86_64-linux/)  { return "linux-x86_64";
     } elsif ($arch =~ m/arm-linux/)     { return "linux-arm";
+    } elsif ($arch =~ m/linux/)        {
+            my($kernel, $hostname, $release, $version, $cpu) = POSIX::uname();
+            if ($cpu =~ m/i686/)			{ return "linux-x86";  }
+            if ($cpu =~ m/x86_64/)	{
+				if ($release =~ m/el5/)     { return "linux-x86_64";  }
+				elsif ($release =~ m/-rt/)  { return "linuxRT-x86_64"; }
+				elsif ($release =~ m/el6/)  { return "rhel6-x86_64"; }
+				elsif ($release =~ m/el7/)  { return "rhel7-x86_64"; }
+				elsif ($release =~ m/2.6.26.1/)  { return "linux-x86_64"; }
+			}
+            else							{ return "unsupported"; }
     } elsif ($arch =~ m/MSWin32-x86/)   { return "win32-x86";
     } elsif ($arch =~ m/MSWin32-x64/)   { return "windows-x64";
     } elsif ($arch =~ m/cygwin/)        {
