diff --git a/configure/CONFIG_SITE b/configure/CONFIG_SITE
index 3063a99..785c0c3 100644
--- a/configure/CONFIG_SITE
+++ b/configure/CONFIG_SITE
@@ -108,0 +109 @@
+#       linuxRT_glibc-x86_64
@@ -111,0 +113,6 @@
+# SLAC PCDS: If EPICS_HOST_ARCH is not set, derive it
+ifndef EPICS_HOST_ARCH
+EPICS_HOST_ARCH	= $(shell $(EPICS_BASE)/startup/EpicsHostArch.pl)
+export EPICS_HOST_ARCH
+endif
+
@@ -117 +124,13 @@ CROSS_COMPILER_TARGET_ARCHS=
-#CROSS_COMPILER_TARGET_ARCHS=vxWorks-ppc32
+
+
+# RHEL6 currently supports linuxRT and RTEMS-beatnik cross compiles
+ifeq '$(EPICS_HOST_ARCH)' 'rhel6-x86_64'
+CROSS_COMPILER_TARGET_ARCHS		+=	linuxRT_glibc-x86_64
+CROSS_COMPILER_TARGET_ARCHS		+=	RTEMS-beatnik
+endif
+
+# RHEL5 currently supports linux-x86 (RHEL5 32bit) and RTEMS-beatnik cross compiles
+ifeq '$(EPICS_HOST_ARCH)' 'linux-x86_64'
+CROSS_COMPILER_TARGET_ARCHS		+=  linux-x86
+CROSS_COMPILER_TARGET_ARCHS		+=	RTEMS-beatnik
+endif
@@ -122,2 +141,8 @@ CROSS_COMPILER_TARGET_ARCHS=
-#
-CROSS_COMPILER_HOST_ARCHS=
+#CROSS_COMPILER_HOST_ARCHS=
+
+###################################
+# PCDS build targets:
+#  host:	linux-x86    targets: RTEMS-beatnik linux-x86
+#  host:	linux-x86_64 targets: RTEMS-beatnik linux-x86 linux-x86_64
+#  host:	rhel6-x86_64 targets: RTEMS-beatnik rhel6-x86_64 linuxRT_glibc-x86_64
+CROSS_COMPILER_HOST_ARCHS		=	linux-x86_64 rhel6-x86_64
@@ -139 +164 @@ SHARED_LIBRARIES=YES
-STATIC_BUILD=NO
+STATIC_BUILD=YES
@@ -188 +213 @@ USE_POSIX_THREAD_PRIORITY_SCHEDULING = YES
-EPICS_SITE_VERSION =
+EPICS_SITE_VERSION = 2.0
@@ -192 +217 @@ EPICS_SITE_VERSION =
-GCC_PIPE = NO
+GCC_PIPE = YES
diff --git a/configure/CONFIG_SITE_ENV b/configure/CONFIG_SITE_ENV
index 75bf1a2..1c7bca8 100644
--- a/configure/CONFIG_SITE_ENV
+++ b/configure/CONFIG_SITE_ENV
@@ -44,0 +45,2 @@
+#	DST for 2016  US: Mar 08 - Nov 01
+#	              EU: Mar 29 - Oct 25
@@ -47,3 +49,2 @@
-# These values are for 2011:
-EPICS_TIMEZONE=CUS::360:031302:110602
-#EPICS_TIMEZONE=MET::-60:032702:103002
+# These values are for 2016:
+EPICS_TIMEZONE=PUS::480:030802:110102
@@ -51 +51,0 @@ EPICS_TIMEZONE=CUS::360:031302:110602
-# EPICS_TS_NTP_INET
@@ -53 +53 @@ EPICS_TIMEZONE=CUS::360:031302:110602
-EPICS_TS_NTP_INET=
+# EPICS_TS_NTP_INET
@@ -61 +61 @@ IOCSH_PS1="epics> "
-IOCSH_HISTSIZE=50
+IOCSH_HISTSIZE=100
