diff --git a/configure/CONFIG_SITE b/configure/CONFIG_SITE
index 785c0c3..4d036b4 100644
--- a/configure/CONFIG_SITE
+++ b/configure/CONFIG_SITE
@@ -112,7 +112,7 @@
 
 # SLAC PCDS: If EPICS_HOST_ARCH is not set, derive it
 ifndef EPICS_HOST_ARCH
-EPICS_HOST_ARCH	= $(shell $(EPICS_BASE)/startup/EpicsHostArch.pl)
+EPICS_HOST_ARCH	= $(shell $(EPICS_BASE)/startup/EpicsHostArch)
 export EPICS_HOST_ARCH
 endif
 
