diff --git a/src/ioc/db/recGbl.c b/src/ioc/db/recGbl.c
index 58baf2a..9154e5d 100644
--- a/src/ioc/db/recGbl.c
+++ b/src/ioc/db/recGbl.c
@@ -268,11 +268,14 @@ void recGblTSELwasModified(struct link *plink)
         errlogPrintf("recGblTSELwasModified called for non PV_LINK\n");
         return;
     }
-    /*If pvname ends in .TIME then just ask for VAL*/
-    /*Note that the VAL value will not be used*/
+    /* pvname must end in .TIME to be recognized as a TSEL time link,
+     * however, we then strip it to make field default to VAL
+     * and set the pvlMask to mark it as a TSEL time link */
     pfieldname = strstr(ppv_link->pvname, ".TIME");
     if (pfieldname) {
         *pfieldname = 0;
+        /* When this TSELisTime flag is set, the VAL field will not be fetched
+         * Instead we'll call dbGetTimeStamp() to just copy the timestamp */
         ppv_link->pvlMask |= pvlOptTSELisTime;
     }
 }
