=== modified file 'src/db/recGbl.c'
--- src/db/recGbl.c	2012-10-17 23:08:39 +0000
+++ src/db/recGbl.c	2013-09-13 16:09:27 +0000
@@ -303,7 +303,7 @@
     /*Note that the VAL value will not be used*/
     pfieldname = strstr(ppv_link->pvname, ".TIME");
     if (pfieldname) {
-        strcpy(pfieldname, ".VAL");
+        *pfieldname = 0;
         ppv_link->pvlMask |= pvlOptTSELisTime;
     }
 }

=== modified file 'src/dbStatic/dbStaticLib.c'
--- src/dbStatic/dbStaticLib.c	2013-05-09 20:48:49 +0000
+++ src/dbStatic/dbStaticLib.c	2013-09-13 16:09:24 +0000
@@ -2041,9 +2041,11 @@
 		else if(pvlMask&pvlOptCP) ppind=3;
 		else if(pvlMask&pvlOptCPP) ppind=4;
 		else ppind=0;
-		if(plink->value.pv_link.pvname)
+		if(plink->value.pv_link.pvname) {
 		    strcpy(message,plink->value.pv_link.pvname);
-		else
+		    if (pvlMask & pvlOptTSELisTime)
+			strcat(message, ".TIME");
+		} else
 		    strcpy(message,"");
 		strcat(message," ");
 		strcat(message,ppstring[ppind]);

